[Embed API] Python Wrapper

charleslai opened the pull request:
Mostly done, waiting for review

charleslai commented:
haha actually not done yet - was leaving a pr open for comments but it seems like i should wait till i have finished everything first

talwai commented:
Cool thanks, adding a test for the new functionality [here](https://github.com/DataDog/datadogpy/blob/master/tests/integration/api/test_api.py) would also be much appreciated

yannmh commented:
@charleslai is it ready for review ?

charleslai commented:
I think I'm good! let me know what needs to be fixed

yannmh commented:
Thanks a lot @charleslai !

Could you clean/concatenate your git commit history to have something short and clear ? Also could you remove the `merge master` commits ? it's making the PR heavier, and the history confusing (prefer a rebase instead) :)

yannmh commented:
Do we plan to bring this new feature to dogshell (aka `dog` command) ?

charleslai commented:
not at the moment I think

charleslai commented:
cleaned git history - add api wrapper and tests

yannmh commented:
Great, I'll review your PR soon. Could you also update the [documentation](https://github.com/DataDog/documentation) to reflect your changes please ?

charleslai commented:
Documentation is updated

yannmh commented on datadog/api/graphs.py:
```diff
@@ -45,3 +50,91 @@ def status(cls, snapshot_url):
+
+        :returns: JSON response from HTTP API request
+        """
+        return super(Embed, cls).get_all()
```
Any reason why you redefine the method when no modification is made ?

charleslai replied:
Nope - removing

yannmh commented on datadog/api/graphs.py:
```diff
@@ -45,3 +50,91 @@ def status(cls, snapshot_url):
+        :returns: JSON response from HTTP API request
+        """
+        handle = embed_id + "/enable"
+        return super(Embed, cls).get(handle)
```
Can we modify the `ActionAPIResource` class and `_trigger_action` method to comply with this type of handle please ? That would make more sense I think.

yannmh commented on tests/integration/api/test_api.py:
```diff
@@ -753,5 +753,152 @@ def test_host_muting(self):
+        result = dog.Embed.revoke(embed_id)
+        # Check embed is revoked and that we can't get it again
+        assert "success" in result
+        try:
```
I know we us this pattern a lot, but can we switch to `assertRaises` instead (c.f. https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises).

yannmh commented:
Great work @charleslai ! Thanks these new features.

It's ready to be merged :ship:.I just made a few nitpicky comments, if you want to adress them.

charleslai commented:
I think I addressed all of the comments - I didn't change the `Action..` class because there is a method there called `_trigger_class_action` that I can use and this combined with `_trigger_action` seems to cover most use cases

yannmh commented:
Great work, thanks @charleslai.

Let's merge it and release a new package :rocket:.
